benchmarking/README.md

```diff
@@ -0,0 +1,568 @@
+# NeMo Curator Benchmarking Framework
+
+A comprehensive benchmarking framework for measuring and tracking the performance of NeMo Curator. This tool enables developers to ensure quality and performance by running standardized benchmark scripts in reproducible environments.
+
+## Table of Contents
+
+- [Quick Start](#quick-start)
+- [Concepts](#concepts)
+- [Configuration](#configuration)
+- [Running benchmarks and using the container](#running-benchmarks-and-using-the-container)
+- [Writing Benchmark Scripts](#writing-benchmark-scripts)
+- [Sinks: Custom Reporting & Actions](#sinks-custom-reporting--actions)
+
+---
+
+## Quick Start
+
+**1. Build the Docker image:**
+
+Assuming the working directory is the NeMo Curator repo root dir:
+```bash
+./benchmarking/tools/build_docker.sh
+```
+
+This builds the `curator_benchmarking` image with:
+- CUDA support
+- Python 3.12 environment
+- NeMo Curator from source in repo root dir
+- All NeMo Curator dependencies
+- Benchmarking framework and scripts
+
+Note: you may only need to do this periodically when the environment needs to be updated. See the `--use-host-curator` example below.
+
+**2. Update config:**
+
+Update `results_path`, `artifacts_path`, and `datasets_path` in the YAML config file based on your preferences. In this example, we'll edit the YAML config `./benchmarking/nightly-benchmark.yaml`
+
+```yaml
+results_path: /path/where/results/are/stored
+artifacts_path: /path/where/artifacts/are/stored
+datasets_path: /path/to/datasets
+```
+
+**3. Run benchmarks:**
+
+```bash
+./benchmarking/tools/run.sh --config ./benchmarking/nightly-benchmark.yaml
+```
+
+To run using the Curator sources on the host instead of those in the image, pass the `--use-host-curator` option:
+```bash
+./benchmarking/tools/run.sh --config ./benchmarking/nightly-benchmark.yaml --use-host-curator
+```
+This is especially useful during active development and debugging since it avoids a costly rebuild step.
+
+
+**4. View results:**
+
+Results are written to the `results_path` specified in your configuration, organized by session timestamp.
+
+---
+
+## Concepts
+
+### Session
+
+A **session** represents a single invocation of the benchmarking framework. Each session:
+- Has a unique name with timestamp (e.g., `nightly-run__2025-01-23__14-30-00`)
```

Comment:
i think default is `benchmark-run__2025-01-23__14-30-00`

Reply:
thanks, updated.